### PR TITLE
Update and audit tutorial 9

### LINKED ISSUE #759: HF audit zkApp tutorials: verify and run through code for Tutorial 9: Recursion
HF audit zkApp tutorials: verify and run through code for Tutorial 9: Recursion

New issue to verify that the code runs for this tutorial as part of the tutorial audit
Earlier issues may not have satisfied the Definition of Done (DoD) where the code is run against the latest versions of o1js and zkApp CLI

### PR DESCRIPTION
**Pull request by ymekuria**
closes #759

### CODE REVIEW

File: docs/zkapps/tutorials/05-common-types-and-functions.mdx [hunk 1]
```diff
@@ -39,6 +39,7 @@ The [example project](https://github.com/o1-labs/docs2/tree/main/examples/zkapps
 
 ## Prerequisites
 
+<<<<<<< HEAD
```
Comment (shimkiv) [inline docs/zkapps/tutorials/05-common-types-and-functions.mdx:42]: Typo I guess

File: docs/zkapps/tutorials/05-common-types-and-functions.mdx [hunk 2]
```diff
@@ -39,6 +39,7 @@ The [example project](https://github.com/o1-labs/docs2/tree/main/examples/zkapps
...
 This tutorial has been tested with [zkApp CLI](https://github.com/o1-labs/zkapp-cli) version `0.20.1` and [o1js](https://www.npmjs.com/package/o1js) version `1.1.0`.
```
Comment (shimkiv) [inline docs/zkapps/tutorials/05-common-types-and-functions.mdx:43]: Let's use link to NPM package as we do for o1js: https://www.npmjs.com/package/zkapp-cli

**Review by shimkiv — CHANGES_REQUESTED** [review]

File: docs/zkapps/tutorials/05-common-types-and-functions.mdx (hunk 2 above)
Comment (shimkiv) [reply docs/zkapps/tutorials/05-common-types-and-functions.mdx:43]: Or it can be handled as part of https://github.com/o1-labs/docs2/issues/926

Comment (ymekuria) [reply docs/zkapps/tutorials/05-common-types-and-functions.mdx]: Let's update these all at once in another pr.

**Review by shimkiv — APPROVED** [review]

**Pull request merged**